### ISSUE TITLE
Padding/Margin of dictionary elements

### ISSUE THREAD

**Issue by facelessuser** [squidfunk/mkdocs-material#173]
## Description

It seems dictionary elements don't have any padding or margin at their bottom when in normal vieweing mode, but when in mobile view, we get good padding.

Normal wide viewing mode:

![screenshot 2017-02-18 09 07 12](https://cloud.githubusercontent.com/assets/1055125/23094657/5c61631e-f5ba-11e6-8cee-a29b9266f3dc.png)

Mobile viewing (narrow mode).

![screenshot 2017-02-18 09 09 15](https://cloud.githubusercontent.com/assets/1055125/23094661/73faa13e-f5ba-11e6-9448-7d4862ef36c3.png)

### Expected behavior

In normal viewing mode, dictionaries will have good looking margin and/or padding at bottom.

### Actual behavior

During normal viewing mode dictionaries have no padding and/or margin at the bottom.

### Steps to reproduce the bug

1. Create two dictionaries after each other in documents
2. Render documents
3. View in normal, wide viewing and mobile, narrow viewing and compare.

### Package versions

* Python: Any
* MkDocs: 0.16.1
* Material: 1.0.5

### Project configuration

Relevant portion:

``` yaml
markdown_extensions:
  - markdown.extensions.toc:
      slugify: !!python/name:pymdownx.slugs.uslugify
      permalink: 
  - markdown.extensions.admonition:
  - markdown.extensions.smarty:
      smart_quotes: false
  - pymdownx.betterem:
  - markdown.extensions.attr_list:
  - markdown.extensions.def_list:
  - markdown.extensions.tables:
  - markdown.extensions.abbr:
  - pymdownx.extrarawhtml:
  - pymdownx.superfences:
      css_class: codehilite
  - pymdownx.highlight:
      extend_pygments_lang:
        - name: php-inline
          lang: php
          options:
            startinline: true
  - pymdownx.inlinehilite:
      css_class: codehilite
  - pymdownx.magiclink:
      repo_url_shortener: true
      base_repo_url: https://github.com/facelessuser/pymdown-extensions
  - pymdownx.tilde:
  - pymdownx.caret:
  - pymdownx.smartsymbols:
  - pymdownx.emoji:
      emoji_generator: !!python/name:pymdownx.emoji.to_svg
  - pymdownx.escapeall:
      hardbreak: True
      nbsp: True
  - pymdownx.tasklist:
      custom_checkbox: true
  - pymdownx.progressbar:
  - pymdownx.arithmatex:
  - pymdownx.mark:
  - pymdownx.plainhtml:
      strip_attributes: ''
  - pymdownx.snippets:
      base_path: docs/snippets
```

### System information

* OS: Don't think it matters, but tested on MacOS
* Browser: Not sure if this is relevant, but Chrome: 56.0.2924.87 (64-bit)

**Comment by squidfunk** [comment]
Thanks for reporting. Can you provide some sample Markdown which causes the issue?

**Comment by facelessuser** [comment]
Yeah.  The key is to have `markdown.extensions.def_list` enabled.

Here's a simple example.  We will do two dictionaries, each with a list (there is a space after the colons).

```
Dictionary 1
: 
    - list

Dictionary 2
: 
    - list
```

When you inspect the HTML in your browser, the `ul` tag will have no margin at the bottom (assuming still rendered in full, wide mode). But if you inspect in mobile mode, you will have a margin of about 16.  This seems to affect many elements like code blocks (as shown in the opening post) etc.

**Comment by facelessuser** [comment]
Wide mode:

![screenshot 2017-02-18 20 33 29](https://cloud.githubusercontent.com/assets/1055125/23099000/9a70930a-f619-11e6-98a4-fe1e981082d4.png)

---

Mobile:

![screenshot 2017-02-18 20 33 15](https://cloud.githubusercontent.com/assets/1055125/23099003/a1787ba4-f619-11e6-8164-3864d722d390.png)

**Comment by squidfunk** [comment]
First of all, thanks for the very detailed explanation! It's a pleasure to work with such a detailed bug report.

The margin bug came from the reset of the bottom margin of the last element within `md-content__inner` which was not correctly scoped, as it applied to every `:last-child` below `md-content__inner`. This reset is necessary due to margin-collapse of the parent element, so that everything aligns with the sidebars. I re-scoped it with the child selector `>`.

I furthermore added some styles for definition lists - left them out originally as I wanted to wait if someone even uses them. I aligned the lists with the normal lists and added a section on the specimen page:

<img width="703" alt="bildschirmfoto 2017-02-19 um 16 17 14" src="https://cloud.githubusercontent.com/assets/932156/23103652/f2d08abe-f6be-11e6-822a-6070831a965f.png">

As I just released 1.0.4 three days ago and 1.0.5 yesterday fixing some nasty bugs, I will wait for some more stuff to accumulate before releasing it because I consider this a rather minor issue, hope that's okay for you.

**Comment by squidfunk** [comment]
I will re-open this issue and leave it open until it is released.

**Comment by facelessuser** [comment]
Nice, no worries, I'll probably just pick up the changes in my extra CSS until this gets released.  Thanks for getting a fix so quickly!

**Comment by squidfunk** [comment]
Fix released in 1.1.0
